server/src/storage/segment.rs

```diff
@@ -88,6 +88,11 @@ impl SegmentBuilder {
 
     /// Add the timestamp and value from `data_point` to the segment's array builders.
     pub fn insert_data(&mut self, data_point: &DataPoint) {
+        debug_assert!(
+            !self.is_full(),
+            "Cannot insert data into full segment builder."
```

Comment:
I also have no strong opinions regarding this. Since doc comments are (potentially) more user-facing, would that mean we would use "segment builder" in doc comments and "SegmentBuilder" in normal comments and "internal" documentation like `debug_assert!()`.

Reply:
Agreed to use the actual types, primarily to support [links](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html).

Reply:
I agree that using the actual type names seems like a good choice here.